contracts/governance/Governor.sol

```diff
@@ -333,59 +336,133 @@ abstract contract Governor is Context, ERC165, EIP712, Nonces, IGovernor, IERC72
     }
 
     /**
-     * @dev See {IGovernor-execute}.
+     * @dev See {IGovernor-queue}.
      */
-    function execute(
+    function queue(
         address[] memory targets,
         uint256[] memory values,
         bytes[] memory calldatas,
         bytes32 descriptionHash
-    ) public payable virtual override returns (uint256) {
+    ) public virtual override returns (uint256) {
+        return _queue(targets, values, calldatas, descriptionHash);
+    }
+
+    /**
+     * @dev Internal queuing mechanism. Can be overridden (with a super call) to add logic before or after the
+     * queuing.
+     *
+     * Emits a {IGovernor-ProposalQueued} event if the eta is not 0.
+     */
+    function _queue(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) internal virtual returns (uint256) {
         uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);
 
-        ProposalState currentState = state(proposalId);
-        if (currentState != ProposalState.Succeeded && currentState != ProposalState.Queued) {
-            revert GovernorUnexpectedProposalState(
-                proposalId,
-                currentState,
-                _encodeStateBitmap(ProposalState.Succeeded) | _encodeStateBitmap(ProposalState.Queued)
-            );
-        }
-        _proposals[proposalId].executed = true;
+        _validateStateBitmap(proposalId, _encodeStateBitmap(ProposalState.Succeeded));
 
-        emit ProposalExecuted(proposalId);
+        (bool queued, uint48 eta) = _doQueue(proposalId, targets, values, calldatas, descriptionHash);
 
-        _beforeExecute(proposalId, targets, values, calldatas, descriptionHash);
-        _execute(proposalId, targets, values, calldatas, descriptionHash);
-        _afterExecute(proposalId, targets, values, calldatas, descriptionHash);
+        if (queued) {
+            _proposalsExtra[proposalId].eta = eta;
```

Comment:
I believe we may need `require(eta > 0)`. We use this assumption in `state()`, if I'm not mistaken?

Reply:
I suppose if `eta == 0` the `queue` operation is a bit of a no-op because it would remain in `Succeeded` state.

Perhaps there is no strong reason not to allow this, other than as a sanity check.

Reply:
If we require eta > 0, does it make sens to remove the `bool queued` and to have `_doQueue` only return a uint32, with 
```
        if (eta > 0) {
            _proposalsExtra[proposalId].eta = eta;
            emit ProposalQueued(proposalId, eta);
        } else {
            revert GovernorQueueNotImplemented();
        }
 ...

Reply:
I think that makes sense. Initially I wanted to allow eta = 0 because because I thought it was a delay, I think. But if it's a timestamp, it should never be 0, we can use it as a natural sentinel value to mean not queued.